mcsema/Arch/Arch.cpp

```diff
@@ -331,6 +342,102 @@ size_t ArchDecodeInstruction(const uint8_t *bytes, const uint8_t *bytes_end,
   return total_size;
 }
 
+// Convert the given assembly instruction into an inline ASM operation in lieu
+// of decompiling it.
+void ArchBuildInlineAsm(llvm::MCInst &inst, llvm::BasicBlock *block) {
+  auto opcode = inst.getOpcode();
+
+  // Use the printer to build the ASM string. We'll need to escape the $ in
+  // register names with $$.
+  std::string asmString, constraints;
```

Comment:
I suggest using a `std::stringstream` here, then calling the `.str()` method when you need the final `std::string`.